servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java

```diff
@@ -124,7 +124,12 @@ protected void assignReference(NamedRef ref, String oldHash, Reference assignTo)
   @Override
   protected void deleteReference(NamedRef ref, String hash)
       throws NessieConflictException, NessieNotFoundException {
-    startAccessCheck().canDeleteReference(ref).checkAndThrow();
+    BatchAccessChecker check = startAccessCheck();
+    if (getConfig().getDefaultBranch().equals(ref.getName())) {
+      check.canDeleteDefaultBranch();
+    }
+    check.canDeleteReference(ref);
```

Comment:
note: we keep the old check in place as well

Reply:
why do we need to check both here? This should be in else case ?

Reply:
Basically for default reference, if the user has access to delete it then it should not matter whether user has delete reference permission or not right?

@nastra, @snazy : WDYT ?

Reply:
moved to else after confirming with Robert